### PR TITLE
Feature/ap 12

### PR DESCRIPTION
**Pull request by rdhawladar**

### CODE REVIEW

File: internal/app/parcel/repository.go [hunk 1]
```diff
@@ -44,3 +46,17 @@ func (r *repository) InsertParcel(ctx context.Context, parcel model.Parcel) erro
...
 }
+
+func (r *repository) FetchParcelByID(ctx context.Context, parcelID int) (model.Parcel, error) {
+	var parcel model.Parcel
+
+	if err := r.db.GetContext(ctx, &parcel, fetchParcelByIDQuery, parcelID); err != nil {
+		if err == sql.ErrNoRows {
+			return model.Parcel{}, fmt.Errorf("parcel %d is not found. :%w", parcelID, model.ErrNotFound)
+		}
+
```
Comment (shuza) [inline internal/app/parcel/repository.go:59]: log this error

File: internal/app/parcel/service.go [hunk 2]
```diff
@@ -28,3 +28,13 @@ func (s *service) CreateParcel(ctx context.Context, parcel model.Parcel) error {
...
 	return s.repo.InsertParcel(ctx, parcel)
 }
+
+func (s *service) GetParcelByID(ctx context.Context, parcelID int) (model.Parcel, error) {
+	parcel, err := s.repo.FetchParcelByID(ctx, parcelID)
+
+	if err != nil {
+		return model.Parcel{}, err
+	}
+
```
Comment (shuza) [inline internal/app/parcel/service.go:41]: log this error

**Review by shuza — APPROVED** [review]
lgtm.. just add error log

File: internal/app/parcel/service.go (hunk 2 above)
Comment (rdhawladar) [reply internal/app/parcel/service.go:41]: added log

**Pull request merged**